homeassistant/components/nest/device_trigger.py

```diff
@@ -33,6 +33,7 @@ async def async_get_nest_device_id(hass: HomeAssistant, device_id: str) -> str |
     device = device_registry.async_get(device_id)
     for (domain, unique_id) in device.identifiers:
         if domain == DOMAIN:
+            assert isinstance(unique_id, str)
```

Comment:
asserts don't work at runtime, what is this checking here?

Reply:
I'm new to mypy and static types. I'm just trying to get this to know it's a string and was following this
https://mypy.readthedocs.io/en/stable/casts.html#casts-and-type-assertions example.

Would it be more common or more appropriate to cast here?

Reply:
I'm not sure, why does it need to be caused? the unique_id is a string already, right?

Reply:
Thanks for pushing this, incorrectly assumed it was having trouble with the tuple, but then doing some guess and check type adding led me to realize it was the DeviceRegistry. Adding a type hint there seemed better. I assume this is from `hass.helpers` but did not look closer.

Fixing this caused it to realize that `device` may be `None` so I added a check for that. If i remember right, that `None` case was difficult to exercise here in a test, so my fix here is playing some games with code coverage.